edgelet/edgelet-docker/src/runtime.rs

```diff
@@ -434,23 +435,74 @@ impl ModuleRuntime for DockerModuleRuntime {
             return Box::new(future::err(Error::from(err)));
         }
 
+        let client = self.client.clone();
+
         Box::new(
-            self.client
+            client
                 .container_api()
                 .container_start(&id, "")
-                .then(|result| match result {
+                .then(move |result| match result {
                     Ok(_) => {
                         info!("Successfully started module {}", id);
-                        Ok(())
+                        future::Either::A(future::ok(()))
                     }
+
                     Err(err) => {
+                        // When using Windows containers, if Moby is restarted after exiting uncleanly, existing containers
+                        // cannot be restarted any more since the container VHDs are held open by `System`.
+                        // Specifically, the start container API endpoint returns an HTTP 500 with the error message:
+                        //
+                        //     hcsshim::ActivateLayer failed in Win32: The process cannot access the file because it is being used by another process. (0x20)
+                        //
+                        // The only way to resolve this is to delete the container and let it be re-created.
+                        // If Edge Agent is deleted, the watchdog will recreate it. For Edge Hub and user modules, Edge Agent will recreate them.
+                        let mut is_hcs_vhd_in_use_by_another_process = false;
+                        if cfg!(windows) {
+                            if let DockerError::Api(err) = &err {
+                                if err.code == StatusCode::INTERNAL_SERVER_ERROR {
+                                    if let Some(message) = crate::error::get_message(err) {
+                                        is_hcs_vhd_in_use_by_another_process = message.starts_with(
+                                            "hcsshim::ActivateLayer failed in Win32: The process cannot access the file because it is being used by another process. (0x20)",
```

Comment:
Does this work on non-en-US Windows? The "The process cannot access the file because it is being used by another process." part comes from `FormatMessage`, so probably not.

Reply:
Actually, golang [hardcodes using en-US when calling `FormatMessage`.](https://github.com/golang/go/blob/5f45a3337ec78f303fbbcadd89d459af56183724/src/syscall/syscall_windows.go#L101) So this will work but is brittle. Probably better to just match the numeric code from the end, per [the pattern used by `hcsshim`.](https://github.com/microsoft/hcsshim/blob/e38a39c0716c28df758f4a8afadc76ec5ceaa165/internal/hcserror/hcserror.go#L21)